src/main/java/network/brightspots/rcv/CommonDataFormatReader.java

```diff
@@ -118,12 +189,205 @@
     return rankings;
   }
 
-  // parse the given file into a List of CastVoteRecords for tabulation
+  // helper to extract the required CVRContest from the input CVR
+  CVRContest getCvrContest(CVR cvr, Contest contestToTabulate) {
+    CVRContest cvrContestToTabulate = null;
+    // find current snapshot
+    CVRSnapshot currentSnapshot = null;
+    if (cvr.CVRSnapshot != null) {
+      for (CVRSnapshot cvrSnapshot : cvr.CVRSnapshot) {
+        if (cvrSnapshot.ObjectId.equals(cvr.CurrentSnapshotId)) {
+          currentSnapshot = cvrSnapshot;
+          break;
+        }
+      }
+    }
+    // find CVRContest which matches the Contest we are tabulating
+    if (currentSnapshot != null && currentSnapshot.CVRContest != null) {
+      for (CVRContest cvrContest : currentSnapshot.CVRContest) {
+        if (cvrContest.ContestId.equals(contestToTabulate.ObjectId)) {
+          cvrContestToTabulate = cvrContest;
+          break;
+        }
+      }
+    }
+    return cvrContestToTabulate;
+  }
+
+  void parseXml(List<CastVoteRecord> castVoteRecords) {
+    try {
+
+      // load XML
+      XmlMapper xmlMapper = new XmlMapper();
+      xmlMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+      FileInputStream inputStream = new FileInputStream(new File(filePath));
+      CastVoteRecordReport cvrReport = xmlMapper.readValue(inputStream, CastVoteRecordReport.class);
+
+      // Find the contest for tabulation
+      Contest contestToTabulate = null;
+      for (Election election : cvrReport.Election) {
+        for (Contest contest : election.Contest) {
+          if (contest.Name.equals(this.contestId)) {
+            contestToTabulate = contest;
+            break;
+          }
+        }
+      }
```

Comment:
Just from reading the code, it's hard for me to figure out what the difference is between this block and the `getCvrContest` method. Could you clarify by adding more verbose comments and/or naming things more descriptively?

Reply:
I think this is the first time you've asked me to add comments.  Very exciting :)